scripts/ZipToGraphics/zipToGraphics.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+'''
+zipToGraphics
+----------------------------------
+Given a .zip file with a shapefile it validates and then returns the feature as a graphic
+'''
+from zipfile import ZipFile
+from os.path import join
+import arcpy
+
+
+required_files = ['shp', 'dbf', 'prj', 'shx']
+wgs = arcpy.SpatialReference(3857)
+
+multiple_warning = 'warning:Multiple features were found in the uploaded shapefile but only the first was returned.'
+categories = {
+    'Terrestrial Treatment Area': ['Polygon'],
+    'Aquatic/Riparian Treatment Area': ['Polygon'],
+    'Affected Area': ['Polygon'],
+    'Easement/Acquisition': ['Polygon'],
+    'Guzzler': ['Point', 'Multipoint'],
+    'Trough': ['Point', 'Multipoint'],
+    'Water control structure': ['Point', 'Multipoint'],
+    'Other point feature': ['Point', 'Multipoint'],
+    'Fish passage structure': ['Point', 'Multipoint'],
+    'Fence': ['Polyline'],
+    'Pipeline': ['Polyline'],
+    'Dam': ['Polyline'],
+    'Research': ['Polygon']
+}
+
+def main(zfilepath, category):
+    messages = []
+
+    # open zip file and get paths
+    zfile = ZipFile(zfilepath)
+    zfilenames = zfile.namelist()
+    zfile_exts = [name.split('.')[1] for name in zfilenames]
+    zfile_name = zfilenames[0].split('.')[0]
```

Comment:
[basename](https://docs.python.org/2/library/os.path.html#os.path.basename) can be helpful. No need to change it unless you want to. So mainly for and FYI.